packages/flutter_tools/lib/src/commands/apk.dart

```diff
@@ -179,227 +191,343 @@ class ApkCommand extends FlutterCommand {
+int _buildApk(_ApkComponents components, String flxPath,
+              ApkKeystoreInfo keystore, String outputFile) {
+  Directory tempDir = Directory.systemTemp.createTempSync('flutter_tools');
+  try {
+    _ApkBuilder builder = new _ApkBuilder(components.androidSdk.path);
 
-      File servicesConfig = new File('${tempDir.path}/services.json');
-      _generateServicesConfig(servicesConfig, components.services);
+    File classesDex = new File('${tempDir.path}/classes.dex');
+    builder.compileClassesDex(classesDex, components.jars);
 
-      _AssetBuilder assetBuilder = new _AssetBuilder(tempDir, 'assets');
-      assetBuilder.add(components.icuData, 'icudtl.dat');
-      assetBuilder.add(new File(flxPath), 'app.flx');
-      assetBuilder.add(servicesConfig, 'services.json');
+    File servicesConfig = new File('${tempDir.path}/services.json');
+    _generateServicesConfig(servicesConfig, components.services);
 
-      _AssetBuilder artifactBuilder = new _AssetBuilder(tempDir, 'artifacts');
-      artifactBuilder.add(classesDex, 'classes.dex');
-      artifactBuilder.add(components.libSkyShell, 'lib/armeabi-v7a/libsky_shell.so');
+    _AssetBuilder assetBuilder = new _AssetBuilder(tempDir, 'assets');
+    assetBuilder.add(components.icuData, 'icudtl.dat');
+    assetBuilder.add(new File(flxPath), 'app.flx');
+    assetBuilder.add(servicesConfig, 'services.json');
 
-      File unalignedApk = new File('${tempDir.path}/app.apk.unaligned');
-      builder.package(unalignedApk, components.manifest, assetBuilder.directory,
-                      artifactBuilder.directory, components.resources);
+    _AssetBuilder artifactBuilder = new _AssetBuilder(tempDir, 'artifacts');
+    artifactBuilder.add(classesDex, 'classes.dex');
+    artifactBuilder.add(components.libSkyShell, 'lib/armeabi-v7a/libsky_shell.so');
 
-      int signResult = _signApk(builder, components, unalignedApk);
-      if (signResult != 0)
-        return signResult;
+    File unalignedApk = new File('${tempDir.path}/app.apk.unaligned');
+    builder.package(unalignedApk, components.manifest, assetBuilder.directory,
+                    artifactBuilder.directory, components.resources);
 
-      File finalApk = new File(argResults['output-file']);
-      ensureDirectoryExists(finalApk.path);
-      builder.align(unalignedApk, finalApk);
+    int signResult = _signApk(builder, components, unalignedApk, keystore);
+    if (signResult != 0)
+      return signResult;
 
-      printStatus('APK generated: ${finalApk.path}');
+    File finalApk = new File(outputFile);
+    ensureDirectoryExists(finalApk.path);
+    builder.align(unalignedApk, finalApk);
 
-      return 0;
-    } finally {
-      tempDir.deleteSync(recursive: true);
+    printStatus('APK generated: ${finalApk.path}');
+
+    return 0;
+  } finally {
+    tempDir.deleteSync(recursive: true);
+  }
+}
+
+int _signApk(_ApkBuilder builder, _ApkComponents components, File apk,
+             ApkKeystoreInfo keystoreInfo) {
```

Comment:
I meant to say the end paran - thuswise:

```
int _signApk(
  _ApkBuilder builder, _ApkComponents components, File apk, ApkKeystoreInfo keystoreInfo
) {
```

There's one or two examples in here: https://github.com/flutter/engine/blob/master/sky/specs/style-guide.md.